File: backend/eventticketingplatform/src/main/java/com/nipapager/eventticketingplatform/event/service/EventService.java
package com.nipapager.eventticketingplatform.event.service;

import com.nipapager.eventticketingplatform.event.dto.EventDTO;
import com.nipapager.eventticketingplatform.response.Response;

import java.time.LocalDate;
import java.util.List;

/**
 * Service interface for event operations
 */
public interface EventService {

    Response<EventDTO> createEvent(EventDTO eventDTO);

    Response<List<EventDTO>> getAllEvents();

    Response<EventDTO> getEventById(Long id);

    Response<EventDTO> updateEvent(Long id, EventDTO eventDTO);

    Response<Void> deleteEvent(Long id);

    Response<List<EventDTO>> searchEvents(
            String city,
            Long categoryId,
            LocalDate startDate,
            LocalDate endDate
    );

    Response<EventDTO> approveEvent(Long id);

    Response<EventDTO> rejectEvent(Long id);

    Response<List<EventDTO>> getMyEvents();
}

File: frontend/event-ticketing-app/src/api/eventService.ts
import api from './axios';
import type { Event } from '../types';

const eventService = {
  // Get all approved events
  getAllEvents: async (): Promise<Event[]> => {
    const response = await api.get('/events');
    return response.data.data;
  },

  // Get event by ID
  getEventById: async (id: number): Promise<Event> => {
    const response = await api.get(`/events/${id}`);
    return response.data.data;
  },

  // Search events with filters
  searchEvents: async (params: {
    search?: string;
    city?: string;
    categoryId?: number;
    startDate?: string;
    endDate?: string;
  }): Promise<Event[]> => {
    const response = await api.get('/events/search', { params });
    return response.data.data;
  },

  // === ORGANIZER FUNCTIONS ===

  // Create new event (Organizer/Admin)
  createEvent: async (eventData: {
    title: string;
    description: string;
    categoryId: number;
    venueId: number;
    eventDate: string;
    eventTime: string;
    imageUrl?: string;
  }): Promise<Event> => {
    const response = await api.post('/events', eventData);
    return response.data.data;
  },

  // Get organizer's events
  getMyEvents: async (): Promise<Event[]> => {
    const response = await api.get('/events/my-events');
    return response.data.data;
  },

  // Update event
  updateEvent: async (id: number, eventData: Partial<Event>): Promise<Event> => {
    const response = await api.put(`/events/${id}`, eventData);
    return response.data.data;
  },

  // Delete event (soft delete - cancel)
  deleteEvent: async (id: number): Promise<void> => {
    const response = await api.delete(`/events/${id}`);
    return response.data;
  },

  // === ADMIN FUNCTIONS ===

  // Approve event (Admin only)
  approveEvent: async (id: number): Promise<Event> => {
    const response = await api.put(`/events/${id}/approve`);
    return response.data.data;
  },

  // Reject event (Admin only)
  rejectEvent: async (id: number): Promise<Event> => {
    const response = await api.put(`/events/${id}/reject`);
    return response.data.data;
  },
};

export default eventService;